thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -166,10 +166,9 @@ class CIndexReadSlaveBase : public CSlaveActivity
             unsigned projectedFormatCrc = helper->getProjectedFormatCrc();
             IOutputMetaData *projectedFormat = helper->queryProjectedDiskRecordSize();
 
-            unsigned p = partNum;
-            while (p<partDescs.ordinality()) // will process all parts if localMerge
+            for (unsigned p = partNum; p<partDescs.ordinality(); p++) // will process all parts if localMerge
             {
-                IPartDescriptor &part = partDescs.item(p++);
+                IPartDescriptor &part = partDescs.item(p);
```

Comment:
The changes in lines 169-171 is so that the correct contextLogger can be referenced with [p].  If p++ remains here then  contextLoggers[p] cannot be used in line 306.